components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2415,6 +2655,50 @@ private static JWT encryptWithPublicKey(Key publicKey, JWTClaimsSet jwtClaimsSet
         }
     }
 
+    /**
+     * Encrypt the JWT token with with given public key.
+     *
+     * @param publicKey           public key used to encrypt
+     * @param signedJwt           contains signed JWT body
+     * @param encryptionAlgorithm JWT signing algorithm
+     * @param spTenantDomain      Service provider tenant domain
+     * @param clientId            ID of the client
+     * @param kid                 value used as 'kid'
+     * @return encrypted JWT token
+     * @throws IdentityOAuth2Exception
+     */
+    private static JWT encryptWithPublicKey(Key publicKey, SignedJWT signedJwt,
+                                            JWEAlgorithm encryptionAlgorithm, EncryptionMethod encryptionMethod,
+                                            String spTenantDomain, String clientId,
+                                            String kid) throws IdentityOAuth2Exception {
+
+        JWEHeader.Builder headerBuilder = new JWEHeader.Builder(encryptionAlgorithm, encryptionMethod);
+
+        try {
+            if (StringUtils.isNotBlank(kid)) {
+                headerBuilder.keyID(kid);
+            }
+            headerBuilder.contentType("JWT"); // required to indicate nested JWT
```

Comment:
```suggestion
            headerBuilder.contentType("JWT"); // Required to indicate nested JWT.
```

Also lets move "JWT" to a constant.

Reply:
Fixed in cd457fe